source/event_stream.c

```diff
@@ -255,25 +260,607 @@ static void s_napi_event_stream_connection_on_connection_shutdown(
     aws_mem_release(shutdown_data->allocator, shutdown_data);
 }
 
+struct aws_event_stream_message_storage {
+    struct aws_allocator *allocator;
+    struct aws_array_list headers;
+    struct aws_byte_buf *payload;
+    enum aws_event_stream_rpc_message_type message_type;
+    uint32_t message_flags;
+};
+
+static void s_aws_event_stream_message_storage_clean_up(struct aws_event_stream_message_storage *storage) {
+    aws_event_stream_headers_list_cleanup(&storage->headers);
+
+    if (storage->payload != NULL) {
+        aws_byte_buf_clean_up(storage->payload);
+        aws_mem_release(storage->allocator, storage->payload);
+    }
+}
+
+static int s_aws_event_stream_message_storage_init_from_native(
+    struct aws_event_stream_message_storage *storage,
+    struct aws_allocator *allocator,
+    const struct aws_event_stream_rpc_message_args *message) {
+
+    storage->allocator = allocator;
+
+    /* we don't use the event stream headers init api so that we can allocate the proper amount */
+    if (aws_array_list_init_dynamic(
+            &storage->headers, allocator, message->headers_count, sizeof(struct aws_event_stream_header_value_pair))) {
+        return AWS_OP_ERR;
+    }
+
+    for (size_t i = 0; i < message->headers_count; ++i) {
+        struct aws_event_stream_header_value_pair *source_header = &message->headers[i];
+
+        if (aws_event_stream_add_header(&storage->headers, source_header)) {
+            goto error;
+        }
+    }
+
+    if (message->payload != NULL) {
+        storage->payload = aws_mem_calloc(allocator, 1, sizeof(struct aws_byte_buf));
+        if (aws_byte_buf_init_copy_from_cursor(
+                storage->payload, allocator, aws_byte_cursor_from_buf(message->payload))) {
+            goto error;
+        }
+    }
+
+    storage->message_type = message->message_type;
+    storage->message_flags = message->message_flags;
+
+    return AWS_OP_SUCCESS;
+
+error:
+
+    /* assumes AWS_ZERO_STRUCT was called first */
```

Comment:
It looks like 'aws_mem_calloc()' on the event that this storage is contained in might be zeroing out all relevant bits but I'm not seeing AWS_ZERO_STRUCT being explicitly called on the storage anywhere prior to this point. My guess is 'aws_mem_calloc()` covers this base but just calling it out in case AWS_ZERO_STRUCT should have been called on the storage specifically.

Reply:
well zeroed memory is zeroed memory